x/staking/keeper/msg_server.go

```diff
@@ -18,6 +21,94 @@ type msgServer struct {
 	Keeper
 }
 
+func (k msgServer) CancelUnbondingDelegation(goCtx context.Context, msg *types.MsgCancelUnbondingDelegation) (*types.MsgCancelUnbondingDelegationResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	valAddr, err := sdk.ValAddressFromBech32(msg.ValidatorAddress)
+	if err != nil {
+		return nil, err
+	}
+	delegatorAddress, err := sdk.AccAddressFromBech32(msg.DelegatorAddress)
+	if err != nil {
+		return nil, err
+	}
+	bondDenom := k.BondDenom(ctx)
+	if msg.Amount.Denom != bondDenom {
+		return nil, sdkerrors.Wrapf(
+			sdkerrors.ErrInvalidRequest, "invalid coin denomination: got %s, expected %s", msg.Amount.Denom, bondDenom,
+		)
+	}
+
+	ubd, found := k.GetUnbondingDelegation(ctx, delegatorAddress, valAddr)
+	if !found {
+		return nil, status.Errorf(
+			codes.NotFound,
+			"unbonding delegation with delegator %s not found for validator %s",
+			msg.DelegatorAddress, msg.ValidatorAddress)
+	}
+
+	var foundUnBondingAtHeight = false
+	var unbondEntry types.UnbondingDelegationEntry
+	var unbondEntryIndex int64
```

Comment:
nit: group these under a single `var ( )` group

```suggestion
	var foundUnBondingAtHeight = false
	var unbondEntry types.UnbondingDelegationEntry
	var unbondEntryIndex int64
```